File: RandomNumber/src/main/Generator.java
package main;

import java.util.Scanner;
import static java.lang.System.*;
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Generator {

    public static void main(String[] args) {
        Scanner input = new Scanner(in);
        Pattern num = Pattern.compile("[0-9]+");
        out.println("Hello, this is your everyday, everyman random number generator!\nLet's get started!");
        while (true) {
            out.print("Pick a number range (no less than 0, no greater than 1,000,000; ex. 1-25): ");
            String range = input.nextLine();
            Matcher matcher = num.matcher(range);
            int a=0;
            int b=0;
            if (matcher.find())
                a = Integer.parseInt(range.substring( matcher.toMatchResult().start(),matcher.toMatchResult().end() ));
            if (matcher.find())
                b = Integer.parseInt(range.substring( matcher.toMatchResult().start(),matcher.toMatchResult().end() ));
            out.printf("Computing...\n");
            int rand = (int)(Math.random()*(b-a)) + a;
            out.printf("You got %d\nAsk again? (Y/n) ", rand);
            if (input.nextLine().toLowerCase().contains("n")) break;
            out.print("Same range? ");
        }
    }
}

File: RandomNumber/src/Generator.kt
import java.lang.System.*
import java.util.*

object Generator {
    val input = Scanner(`in`)
    var a: Int? = 0
    var b: Int? = 0

    @JvmStatic
    fun main(args: Array<String>) {
        val num = Regex("[0-9]+")
        println("Hello, this is your everyday, everyman random number generator!\nLet's get started!\n")
        while (true) {
            print("Pick a number range (no less than 0, no greater than 1,000,000; ex. 1-25): ")
            val range = input.nextLine()
            val result: Sequence<MatchResult>? = num.findAll(range)

            if (Regex("^[nN]").containsMatchIn(range))
                break

            result?.let {
                a = it.elementAtOrNull(0)?.value?.toInt()
                b = it.elementAtOrNull(1)?.value?.toInt()
            }

            if (!compute()) break
        }
    }

    fun compute(): Boolean {
        println("Computing...\n")

        if (a != null && b != null) {
            println("\nThe number is ${Compute(a as Int, b as Int).next()}\n")
            //println("\nThe number is ${Compute(a as Int, b as Int).genList}\n")

            //Asks whether to go again, or quit the program
            print("Ask again? (Y/n) ")
            if (Regex("^[nN]").containsMatchIn(input.nextLine()))
                return false

            //Asks whether to use same number range, or ask a new range
            print("Same range? (Y/n) ")
            if (Regex("^[yY]").containsMatchIn(input.nextLine()))
                compute()
            else return true
        }
        else {
            println("Invalid")
            return true
        }
        return false
    }

}
